WooCommerce/Classes/ViewRelated/Orders/Order Details/Order Summary Section/Edit Order Status/OrderStatusListViewController.swift

```diff
@@ -162,89 +119,44 @@ extension OrderStatusListViewController {
-    }
-}
-
-
-// MARK: - Update the Order status
-//
-private extension OrderStatusListViewController {
-    /// Dispatches an Action to update the order status
-    ///
-    private func updateOrderStatus() {
-        guard let newStatus = selectedStatus?.status else {
+        guard let indexOfSelectedStatus = indexOfSelectedStatus else {
+            didSelectCancel?()
             return
         }
-
-        let orderID = order.orderID
-        let undoStatus = order.status
-
-        let done = updateOrderAction(siteID: order.siteID, orderID: orderID, status: newStatus)
-        let undo = updateOrderAction(siteID: order.siteID, orderID: orderID, status: undoStatus)
-
-        ServiceLocator.stores.dispatch(done)
-        ServiceLocator.analytics.track(.orderStatusChange,
-                                  withProperties: ["id": orderID,
-                                                   "from": undoStatus.rawValue,
-                                                   "to": newStatus.rawValue])
-
-        displayOrderUpdatedNotice {
-            ServiceLocator.stores.dispatch(undo)
-            ServiceLocator.analytics.track(.orderStatusChange,
-                                      withProperties: ["id": orderID,
-                                                       "from": newStatus.rawValue,
-                                                       "to": undoStatus.rawValue])
+        guard let selectedStatus = viewModel.status(at: indexOfSelectedStatus) else {
+            didSelectCancel?()
+            return
         }
-    }
-
-    /// Returns an Order Update Action that will result in the specified Order Status updated accordingly.
-    ///
-    private func updateOrderAction(siteID: Int64, orderID: Int64, status: OrderStatusEnum) -> Action {
-        return OrderAction.updateOrder(siteID: siteID, orderID: orderID, status: status, onCompletion: { error in
-            guard let error = error else {
-                NotificationCenter.default.post(name: .ordersBadgeReloadRequired, object: nil)
-                ServiceLocator.analytics.track(.orderStatusChangeSuccess)
-                return
-            }
-
-            ServiceLocator.analytics.track(.orderStatusChangeFailed, withError: error)
-            DDLogError("⛔️ Order Update Failure: [\(orderID).status = \(status)]. Error: \(error)")
-
-            self.displayErrorNotice(orderID: orderID)
-        })
+        didSelectApply?(selectedStatus)
     }
 }
 
-
 // MARK: - UITableViewDatasource conformance
 //
 extension OrderStatusListViewController: UITableViewDataSource {
     func numberOfSections(in tableView: UITableView) -> Int {
-        return statusResultsController.sections.count
+        return 1
     }
 
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return statusResultsController.sections[section].numberOfObjects
+        guard section == 0 else {
+            return 0
+        }
```

Comment:
I don't think this is necesary! :-)

Reply:
I'd like to keep it, especially as a protection against future sectioning of statuses